app/clients/sms/twilio.py

```diff
@@ -52,7 +52,7 @@ def send_sms(self, to, content, reference, sender=None):
         try:
             message = self._client.messages.create(
```

Comment:
This should be using message service id that gets passed in the request to Notify: https://www.twilio.com/docs/messaging/services 

```

 def send_sms(self, to, content, reference, sender=None):
        # could potentially select from potential numbers like this
        # from_number = random.choice(self._client.incoming_phone_numbers.list()).phone_number
        start_time = monotonic()        
        callback_url = "{}/notifications/sms/twilio/{}".format(
            self._callback_notify_url_host, reference) if self._callback_notify_url_host else ""
        try:
			if (sender is None):
				message = self._client.messages.create(
					to=to,
					from_=from_number,
					body=content,
					status_callback=callback_url,
				)
			else:
				 message = self._client.messages.create(
					to=to,
					body=content,
					messaging_service_sid=sender.message_service_id
					status_callback=callback_url
				)
            self.logger.info("Twilio send SMS request for {} succeeded: {}".format(reference, message.sid))
        except Exception as e:
            self.logger.error("Twilio send SMS request for {} failed".format(reference))
            raise e
        finally:
            elapsed_time = monotonic() - start_time
self.logger.info("Twilio send SMS request for {} finished in {}".format(reference, elapsed_time))

```

Reply:
message_service_id, if available, is passed to send_sms as the "sender" parameter upstream in app/delivery/send_to_providers.py::send_sms_to_provider lines 82-95 (after these changes).

Reply:
right, this is the twilio.py file.  when you get that message service id, you need to create a new twilio message that specifies that message service id.  I saw the changes in send_sms_to_provider.  I wasnt sure how to properly access that property of sender, so I just made an example like above (sender.message_service_id in the else)